doc/state-component/api.md

```diff
@@ -0,0 +1,123 @@
+
+Questions :
```

Comment:
@gregorylegarec : what about `propTypes`; @m4dz would you know another micro framework to handle this?